lib/rules/at-rule-no-deprecated/index.mjs

```diff
@@ -0,0 +1,64 @@
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { deprecatedAtKeywords } from '../../reference/atKeywords.mjs';
+import isStandardSyntaxAtRule from '../../utils/isStandardSyntaxAtRule.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'at-rule-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule) => `Unexpected deprecated at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-no-deprecated',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAtRules: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkAtRules((atRule) => {
+			if (!isStandardSyntaxAtRule(atRule)) return;
+
+			const { name } = atRule;
+
+			if (optionsMatches(secondaryOptions, 'ignoreAtRules', name)) return;
+
+			if (!deprecatedAtKeywords.has(name.toLowerCase())) return;
+
+			const atName = `@${name}`;
+
+			report({
+				message: messages.rejected,
+				messageArgs: [atName],
+				node: atRule,
+				ruleName,
+				result,
+				word: atName,
```

Comment:
IIRC this is not recommended anymore.
@romainmenke should we continue using `word` internally?

Reply:
```suggestion
				index: 0,
				endIndex: atName.length,
```

This would be better.

------

The `at-rule` `name` is maybe the one instance where `word` is acceptable :)
It is always at `index` `0`, so it is always first and always singular.
Using `word` can only find the first instance of multiple problems.

But calculating position information from `word` is slower as it requires serializing the entire current node.

This might be a large amount of AST nodes as `at-rules` can span many style rules each containing many declarations.

Reply:
Thanks for the explanation. It all makes sense.